libraries/bio/src/utils/units-handler.ts

```diff
@@ -32,16 +28,54 @@ export class UnitsHandler {
   public static readonly RnaFastaAlphabet = new Set<string>(['A', 'C', 'G', 'U']);
 
   public static setUnitsToFastaColumn(col: DG.Column) {
+    if (col.semType !== DG.SEMTYPE.MACROMOLECULE || col.getTag(DG.TAGS.UNITS) != NOTATION.FASTA)
+      throw new Error(`The column of notation '${NOTATION.FASTA}' must be '${DG.SEMTYPE.MACROMOLECULE}'.`);
+
+    col.setTag(DG.TAGS.UNITS, NOTATION.FASTA);
+    UnitsHandler.setTags(col, 5, splitterAsFasta);
+  }
+
+  public static setUnitsToSeparatorColumn(col: DG.Column, separator?: string) {
+    if (col.semType !== DG.SEMTYPE.MACROMOLECULE || col.getTag(DG.TAGS.UNITS) != NOTATION.SEPARATOR || !separator)
+      throw new Error(`The column of notation '${NOTATION.SEPARATOR}' must be '${DG.SEMTYPE.MACROMOLECULE}'.`);
+    if (!separator)
```

Comment:
you have the same condition above
maybe check at first for the separator only, and then the other stuff?